installMaGe.py

```diff
@@ -26,91 +18,128 @@
-
-    print('Please do:')
-    print('source setup_mage.sh')
-    print('python', ' '.join(sys.argv))
+# find CLHEP
+try:
+    clhep_include_dir = os.environ.get('CLHEP_BASE_DIR', None)
+    if clhep_include_dir is None:
+        clhep_include_dir = subprocess.run(['clhep-config','--prefix'], check=True, universal_newlines=True, stdout=subprocess.PIPE).stdout.split('"')[1]
+    clhep_include_dir = clhep_include_dir + '/include'
+except Exception:
+    print("Could not find CLHEP_BASE_PATH or clhep-config")
     sys.exit()
 
+# determine geant4, generate setup_mage.sh, and source it
+print('Generating setup_mage.sh ...')
+g4path = re.sub('data/G4EMLOW.*', '', os.environ['G4LEDATA'])
+with open('setup_mage.sh', 'w') as file:
+    file.write(f'source {g4path}geant4make/geant4make.sh\n')
+    file.write(f'export MGDODIR={pwd}/MGDO\n')
+    file.write(f'export TAMDIR=$MGDODIR/tam\n')
+    file.write(f'export MAGEDIR={pwd}/MaGe/build\n')
+    file.write(f'export MGGENERATORDATA={pwd}/share/MaGe/generators\n')
+    file.write(f'export MPPDIR={pwd}/mage-post-proc\n')
+    file.write(f'export PATH={install_path}/bin:$PATH\n')
+    file.write(f'export LD_LIBRARY_PATH={install_path}/lib:$LD_LIBRARY_PATH\n')
+    file.write(f'export ROOT_INCLUDE_PATH={clhep_include_dir}:' \
+               f'{install_path}/include/mgdo:' \
+               f'{install_path}/include/tam:$TAMDIR:' \
+               f'{install_path}/include/mage:' \
+               f'{install_path}/include/mage-post-proc:' \
+               '${ROOT_INCLUDE_PATH}\n')
+    file.write(f'export PYTHONPATH={install_path}/lib:$PYTHONPATH\n')
+
+# make absolutely sure necessary variables are set for installation
+os.environ['MGDODIR']=pwd+'/MGDO'
+os.environ['PATH']=install_path+'/bin:'+os.environ['PATH']
 
 # download (user will enter password)
 try:
     if not os.path.exists('MGDO'):
-        subprocess.run(['git', 'clone', 'https://github.com/mppmu/MGDO.git'], check=True)
+        subprocess.run(f'git clone -b {args.mgdobranch} https://github.com/{args.mgdofork}/MGDO.git'.split(), check=True)
     if not os.path.exists('MaGe'):
-        subprocess.run(['git', 'clone', 'https://github.com/mppmu/MaGe.git'], check=True)
-    if not os.path.exists('GAT'):
-        subprocess.run(['git', 'clone', 'https://github.com/mppmu/GAT.git'], check=True)
+        subprocess.run(f'git clone -b {args.magebranch} https://github.com/{args.magefork}/MaGe.git MaGe/source'.split(), check=True)
+    if not os.path.exists('mage-post-proc'):
+        subprocess.run(f'git clone -b {args.mppbranch} https://github.com/{args.mppfork}/mage-post-proc.git'.split(), check=True)
 except subprocess.CalledProcessError:
     sys.exit()
-
+    
 # install MGDO
 os.chdir('MGDO')
-subprocess.run(['./configure', '--prefix='+install_path, '--enable-majorana-all'])
+subprocess.run(f'./configure --prefix={install_path} --enable-streamers --enable-tam --enable-tabree'.split())
 subprocess.run(['make'])
 subprocess.run(['make', 'install'])
```

Comment:
what about forcing single-threaded mode here?
```suggestion
subprocess.run(['make', 'install', '-j1'])
```
By the way, why isn't `args.jobs` used here?

Reply:
jobs is not used for MGDO because it doesn't fully compile with -j>1. One option would be to have it do `make -j{args.jobs}`, `make`. I agree about forcing single threaded mode; for me it always defaults to 1 if no `-j` option is given, but if that's not something we can count on we should play it safe.

Reply:
That's right, the default is `-j1`. I forgot I permanently `export MAKEFLAGS="-j$(nproc)"` in my environment...